### PR TITLE
c2c: unskip some tests under stress race

### PR DESCRIPTION
**Pull request by msbutler**
Unskipping:
- TestTenantStreamingPauseOnPermanentJobError
- TestTenantStreamingPauseResumeIngestion
- TestTenantStreamingProducerJobTimedOut
- TestTenantStreamingCheckpoint

Remove redundant TestTenantStreaming test

Epic: None

Release note: None

### CODE REVIEW

File: pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go [hunk 1]
```diff
@@ -229,12 +216,13 @@ func TestTenantStreamingPauseOnPermanentJobError(t *testing.T) {
 	c, cleanup := replicationtestutils.CreateTenantStreamingClusters(ctx, t, args)
 	defer cleanup()
 
-	// Make ingestion error out only once.
+	// Make ingestion error out only once to ensure the job conducts one retryable
+	// error. It's fine to close the channel before the receiver gets the error,
+	// as the channel is buffered.
```
Comment (lidorcarmel) [inline pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go:221]: and if it wasn't buffered it was still fine..
maybe:
It's fine to close the buffered channel, the receiver will still get the error.
or something like that?

File: pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go [hunk 2]
```diff
@@ -340,20 +324,13 @@ func TestTenantStreamingCheckpoint(t *testing.T) {
...
 		require.Less(t, checkpointMinTime.UnixNano(), clientStartTime.GoTime().UnixNano())
 	}
 
-	// After cutover, changes to source won't be streamed into destination cluster.
-	c.SrcExec(func(t *testing.T, sysSQL *sqlutils.SQLRunner, tenantSQL *sqlutils.SQLRunner) {
-		tenantSQL.Exec(t, `INSERT INTO d.t2 VALUES (3);`)
-	})
-	// Check the dst cluster didn't receive the change after a while.
-	<-time.NewTimer(3 * time.Second).C
-	c.RequireDestinationFingerprintAtTimestamp(cutoverFingerprint, c.DestSysServer.Clock().Now().AsOfSystemTime())
```
Comment (lidorcarmel) [inline pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go:356]: as much as I don't like the 3 seconds thing.. isn't this an important part of this test?

**Review by lidorcarmel — APPROVED** [review]

File: pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go (hunk 2 above)
Comment (msbutler) [reply pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go:356]: we already exercise this in a different test [here](https://github.com/cockroachdb/cockroach/blob/bad486497ada5cc5465398c71fe8dedb334e59ef/pkg/ccl/streamingccl/streamingest/testdata/simple#L93). I don't think doing this  again in this test adds more coverage.

File: pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go (hunk 1 above)
Comment (msbutler) [reply pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go]: Done.

**Comment by msbutler** [comment]
unrelated flake

**Comment by msbutler** [comment]
TFTR!

bors r=lidorcarmel

**Comment by msbutler** [comment]
blathers backport 23.1

**Pull request merged**